### ISSUE TITLE
`nix flake lock --help` shows unexepected HTML

### ISSUE THREAD

**Issue by n8henrie** [NixOS/nix#9665]
**Describe the bug**

This may be a home-manager issue? or due to having set `bat` as my pager, but looking at https://github.com/NixOS/nix/blob/e23983db824f9ea0ecdb6cab3739232b2554f2f5/src/nix/main.cc#L204 I can't decide, and overriding `MANPAGER` doesn't seem to fix anything, so I thought I'd start here.

The bottom line is that I see unexpected HTML output in the `--help` of some new-style nix commands, for example:

```console
$ nix flake lock --help | tail
        Enable the use-xdg-base-directories setting.

      · <span id="opt-user-agent-suffix">--user-agent-suffix</span> value

        Set the user-agent-suffix setting.

      · <span id="opt-warn-dirty">--warn-dirty</span>

        Enable the warn-dirty setting.

```

This happens on both MacOS with nix-darwin as well as NixOS, pinned to 23.11 for both.

My default pager is `bat`:

```console
$ echo $MANPAGER
sh -c 'col -bx | bat -l man -p'
```

...but setting MANPAGER to more conventional settings makes no difference:

```console
$ MANPAGER=less nix flake lock --help | tail
        Enable the use-xdg-base-directories setting.

      · <span id="opt-user-agent-suffix">--user-agent-suffix</span> value

        Set the user-agent-suffix setting.

      · <span id="opt-warn-dirty">--warn-dirty</span>

        Enable the warn-dirty setting.
$ MANPAGER=cat nix flake lock --help | tail
        Enable the use-xdg-base-directories setting.

      · <span id="opt-user-agent-suffix">--user-agent-suffix</span> value

        Set the user-agent-suffix setting.

      · <span id="opt-warn-dirty">--warn-dirty</span>

        Enable the warn-dirty setting.

```

I see the same bug with `nix help flake lock`. However, either of these work as expected (albeit piping to tail gives a ton of `troff:<standard input>:862: warning: cannot select font 'C'` errors):

```console
$ man nix3-flake-lock 
$ MANPAGER=cat man nix3-flake-lock
```

**Steps To Reproduce**

As per above

**Expected behavior**

I expected output / behavior from `nix flake lock --help` to match that of `man nix3-flake-lock`

**`nix-env --version` output**

Same on MacOS and NixOS

```console
$ nix-env --version
nix-env (Nix) 2.18.1
```

**Additional context**

home-manager also pinned to 23.11. 

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by thufschmitt** [comment]
That was indeed an issue with the manpage rendering, but it got fixed between 2.18 and 2.19:

```console
$ nix run nix/2.18.1 --  flake lock --help | tail
      Enable the use-xdg-base-directories setting.

    · <span id="opt-user-agent-suffix">--user-agent-suffix</span> value

      Set the user-agent-suffix setting.

    · <span id="opt-warn-dirty">--warn-dirty</span>

      Enable the warn-dirty setting.

$ nix run nix/2.19.1 --  flake lock --help | tail
      During evaluation, rewrite missing or corrupted files in the Nix store. During building, rebuild missing or corrupted store paths.

    · --version

      Show version information.

    | Note

    | See man nix.conf for overriding configuration settings with command line flags.
```

I'll close this then, feel free to reopen if it still happens with newer versions

**Comment by n8henrie** [comment]
Interesting. It doesn't look like 2.19 is packaged in nixpkgs -- should this be a nixpkgs issue then?

```console
$ nix eval github:nixos/nixpkgs/master#nix.version
"2.18.1"
```

Ah, nevermind -- I see, it's just not in `release-23.11`; using my unstable overlay works fine:  `nix.package = pkgs.unstable.nixVersions.nix_2_19;`, solves the issue.

Thank you!
